CONTRIBUTING.md

```diff
@@ -106,30 +108,43 @@ As mentioned in the previous section, tests are split into groups thanks to [jes
+* `AWS_PROFILE` to use the right AWS credentials
 * `DISABLE_TEARDOWN` if you don't want your stack to be destroyed at the end of the test (useful in dev mode when iterating over your code).
 
-Example: `DISABLE_TEARDOWN=true AWS_PROFILE=ara npx jest --group=integ/other/example`
+Example: `DISABLE_TEARDOWN=true AWS_PROFILE=dev-account npx jest --group=e2e/metrics/decorator`
 
 **Automate**
 
 You can run the end-to-end tests automatically on your forked project by following these steps:
-1. Create AWS Role
-  As mention earlier we are leveraging CDK to deploy and clean resources on AWS. Therefore to run those tests through github actions you will need to grant specific permissions to your workflow. To do so you can leverage [@pahud/cdk-github-oidc](https://constructs.dev/packages/@pahud/cdk-github-oidc) construct which setup the right resources to leverage [Github OpenID Connect](https://github.blog/changelog/2021-10-27-github-actions-secure-cloud-deployments-with-openid-connect/) mechanism.
-1. Add your new role into your Github fork secrets under `AWS_ROLE_ARN_TO_ASSUME`.
-1. Run manually `run-e2e-tests` workflow.
+1. Create an IAM role in your target AWS account, with the least amount of privilege.
+
+    As mentioned above in this page, we are leveraging CDK to deploy and consequently clean-up resources on AWS. Therefore to run those tests through Github actions you will need to grant specific permissions to your workflow.  
+
+    We recommend following [Amazon IAM best practices](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html) for the AWS credentials used in GitHub Actions workflows, including:
+    * Do not store credentials in your repository's code.
+    * [Grant least privilege](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#grant-least-privilege) to the credentials used in GitHub Actions workflows.  Grant only the permissions required to perform the actions in your GitHub Actions workflows.
+    * [Monitor the activity](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#keep-a-log) of the credentials used in GitHub Actions workflows.
+
+    For an example of how to create a role in CDK, you can look at [@pahud/cdk-github-oidc](https://constructs.dev/packages/@pahud/cdk-github-oidc) construct.
+
+    More information about:  
+
+    - [Github OpenID Connect](https://github.blog/changelog/2021-10-27-github-actions-secure-cloud-deployments-with-openid-connect/
+    - ["Configure AWS Credentials" Action For GitHub Actions](https://github.com/aws-actions/configure-aws-credentials/)
+1. Add your new role into your [Github fork secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) with name  `AWS_ROLE_ARN_TO_ASSUME`.
+1. In your forked repository, go to the "Actions" tabs, select the `run-e2e-tests` workflow.
+1. In the run-e2e-tests workflow page, select "Run workflow" and run it on the desired branch.
```

Comment:
Nit: steps numbers are all "1."

Reply:
They rendered correctly by GitHub preview & [according to spec](https://www.markdownguide.org/basic-syntax/#ordered-lists) it's not required for them to be in order:
>  To create an ordered list, add line items with numbers followed by periods. The numbers don’t have to be in numerical order, but the list should start with the number one.

But for those who are reading the source I agree that it was confusing, changed it in [973a9d5](https://github.com/awslabs/aws-lambda-powertools-typescript/pull/441/commits/973a9d5e439e9b0706775177bd65a5d97f9ced3f).